// Copyright 2019 The TensorFlow Authors.
// Licensed under the Apache License, Version 2.0.

syntax = "proto3";
option cc_enable_arenas = true;
package codewithgpu.tensorflow;

// Byte list container.
message BytesList {
  repeated bytes value = 1;
}

// Float list container.
message FloatList {
  repeated float value = 1 [packed = true];
}

// Int64 list container.
message Int64List {
  repeated int64 value = 1 [packed = true];
}

// Feature definition.
message Feature {
  // Each feature can be exactly one kind.
  oneof kind {
    BytesList bytes_list = 1;
    FloatList float_list = 2;
    Int64List int64_list = 3;
  }
};

// Map container.
message Features {
  map<string, Feature> feature = 1;
};

// Example definition.
message Example {
  Features features = 1;
};
